api/k8s/v1/k8s.proto

```diff
@@ -319,6 +319,10 @@ message Pod {
 
 message ListOptions {
   map<string, string> labels = 9;
+  // This selector string will be used in case the labels is empty
```

Comment:
use a oneof?

Reply:
it says that map fields are not allowed in oneofs, because internally they are repeated fields and those are not allowed.  It seems to not complain if I wrap it in a struct.  Is that the way to go?

Reply:
ah that's right. meh, either way is fine i guess. i'll defer to the other thread where you're deciding on whether or not to concat them.